### PR TITLE
UNOMI-492 Make rules engine more robust when some rules are invalid (null actions)

### PR DESCRIPTION
**Pull request by sergehuber**
Here's what was achieved in this PR: 
- Make sure we handle null actions gracefully instead of generating a NPE and blocking loading of further rules
- Add an integration test to test the case of null actions in rule
- Improve logging when rules have null or empty conditions and actions

### CODE REVIEW

**Review by jkevan — CHANGES_REQUESTED** [review]

File: itests/src/test/java/org/apache/unomi/itests/RuleServiceIT.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        LOGGER.info("Waiting for rules to refresh from persistence...");
+        int loopCount = 0;
+        int lastRuleCount = initialRuleCount;
+        while (loopCount < 20 && lastRuleCount == initialRuleCount) {
+            Thread.sleep(1000);
+            ruleMetadatas = rulesService.getRuleMetadatas();
+            lastRuleCount = ruleMetadatas.size();
+            loopCount++;
+        }
+        assertEquals("Rule not properly saved", initialRuleCount + 1, lastRuleCount);
```
[the remark this answers is not in the document]
Comment (sergehuber) [reply itests/src/test/java/org/apache/unomi/itests/RuleServiceIT.java:83]: yes you are right I need to correct that. Thanks for catching it.

[the remark this answers is not in the document]
Comment (sergehuber) [reply itests/src/test/java/org/apache/unomi/itests/RuleServiceIT.java]: Ok I've modified the test to simply use getRule

**Review by jkevan — APPROVED** [review]
